src/slip.rs

```diff
@@ -0,0 +1,200 @@
+use secp256k1::{PublicKey};
+
+/// A UTXO slip containing record of owernship of funds on the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Slip {
+    /// Contains concrete data which is not relative to state of the chain
+    body: SlipBody,
+    /// Flag determining if the `Slip` is part of the longest chain
+    lc: bool,
```

Comment:
I'd like to add this when it becomes relevant, i.e. with mempool because I'm not convinced we need it until I see the algorithm that will use it.

Reply:
Won't we want to call this "spent" rather than "lc" also? let's just tackle this later, there's enough to talk about with just the slip/transaction encoding.